File: app/data/consoles.py
"""
Seed iniziale del catalogo console (tabella `consoles`, popolata una sola
volta al primo avvio da `seed_consoles()` in app/core/bootstrap.py). Dopo il
seed, il catalogo è gestito interamente da superadmin via CRUD
(`app/controllers/utenti/consoles.py`) — questo modulo non è più la fonte di
validazione a runtime (vedi app/services/utenti/ownership.py, che ora legge
dalla tabella `consoles`).
"""

# (chiave stabile usata a DB, etichetta per la UI, R4-compatibile)
SEED_CONSOLES = [
    ("ds", "DS", True),
    ("ds_lite", "DS Lite", True),
    ("dsi", "DSi", True),
    ("dsi_xl", "DSi XL", True),
    ("3ds", "3DS", True),
    ("3ds_xl", "3DS XL", True),
    ("2ds", "2DS", True),
    ("new_3ds", "New 3DS", True),
    ("new_3ds_xl", "New 3DS XL", True),
    ("new_2ds_xl", "New 2DS XL", True),
    ("switch", "Switch", False),
    ("switch_lite", "Switch Lite", False),
    ("switch_oled", "Switch OLED", False),
    ("switch_2", "Switch 2", False),
]

MKDS_GAME_ID = 1  # game_id=1 è sempre Mario Kart DS (vedi app/core/bootstrap.py)

File: frontend/src/lib/consoles.js
// Mirror di app/data/consoles.py — le chiavi devono restare identiche a quelle
// validate lato backend.
export const CONSOLE_LIST = [
    { key: 'ds', label: 'DS' },
    { key: 'ds_lite', label: 'DS Lite' },
    { key: 'dsi', label: 'DSi' },
    { key: 'dsi_xl', label: 'DSi XL' },
    { key: '3ds', label: '3DS' },
    { key: '3ds_xl', label: '3DS XL' },
    { key: '2ds', label: '2DS' },
    { key: 'new_3ds', label: 'New 3DS' },
    { key: 'new_3ds_xl', label: 'New 3DS XL' },
    { key: 'new_2ds_xl', label: 'New 2DS XL' },
    { key: 'switch', label: 'Switch' },
    { key: 'switch_lite', label: 'Switch Lite' },
    { key: 'switch_oled', label: 'Switch OLED' },
    { key: 'switch_2', label: 'Switch 2' },
]

const R4_KEYS = new Set([
    'ds', 'ds_lite', 'dsi', 'dsi_xl', '3ds', '3ds_xl',
    '2ds', 'new_3ds', 'new_3ds_xl', 'new_2ds_xl',
])
export const R4_DEVICE_LIST = CONSOLE_LIST.filter((c) => R4_KEYS.has(c.key))

export const MKDS_GAME_ID = 1

export const consoleLabel = (key) => CONSOLE_LIST.find((c) => c.key === key)?.label ?? key
